aptos-move/move-examples/token_objects/guild/sources/guild.move

```diff
@@ -0,0 +1,321 @@
+///   - a guild collection manager object is created,
+///   - a guild collection is created with the signer of the guild collection manager, and
+///   - a whitelist for guild masters is created.
+/// 2. The admin adds guild masters to the whitelist.
+/// 3. A whitelisted guild master calls `mint_guild` to mint a guild token,
+///    and creates a member collection for the guild.
+/// 4. A guild master with a guild token calls `mint_member` to mint
+///    a member token for a user who is a guild member.
+/// 5. A guild master burns (or revokes) a member token.
+module guild::guild {
+    use std::option;
+    use std::signer;
+    use std::string::{Self, String};
+    use aptos_framework::object::{Self, Object};
+    use aptos_std::smart_vector::{Self, SmartVector};
+    use aptos_token_objects::collection;
+    use aptos_token_objects::token;
+
+    /// The provided signer is not the admin
+    const ENOT_ADMIN: u64 = 1;
+    /// The provided signer is not the owner
+    const ENOT_OWNER: u64 = 2;
+    /// The provided signer is not a guild master
+    const ENOT_GUILD_MASTER: u64 = 3;
+
+    /// The guild token collection name
+    const GUILD_COLLECTION_NAME: vector<u8> = b"Guild Collection Name";
+    /// The guild collection description
+    const GUILD_COLLECTION_DESCRIPTION: vector<u8> = b"Guild Collection Description";
+    /// The guild collection URI
+    const GUILD_COLLECTION_URI: vector<u8> = b"https://guild.collection.uri";
+
+    /// Whitelisted guild masters
+    struct Config has key {
+        whitelist: SmartVector<address>,
+    }
+
+    /// Guild collection manager refs
+    struct Manager has key {
+        /// Used to obtain the signer of the guild collection manager object.
+        extend_ref: object::ExtendRef,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Guild token
+    struct GuildToken has key {
+        /// Used to get the signer of the token
+        extend_ref: object::ExtendRef,
+        /// Member collection name
+        member_collection_name: String,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Member token
+    struct MemberToken has key {
+        /// Belonging guild
+        guild: Object<GuildToken>,
+        /// Used to burn
+        burn_ref: token::BurnRef,
+    }
+
+    /// Initializes the module, creating the manager object, the guild token collection and the whitelist.
+    fun init_module(sender: &signer) acquires Manager {
+        // Create the guild collection manager object to use it to autonomously
+        // manage the guild collection (e.g., create the collection and mint tokens).
+        create_guild_collection_manager(sender);
+        // Create the guild collection
+        create_guild_collection(&guild_collection_manager_signer());
+        // Create a whitelist
+        move_to(sender, Config { whitelist: smart_vector::empty() });
```

Comment:
Can this be combined with the Manager struct? Both are stored on the deployer (sender)'s account anyway

Reply:
Updated. `Manager` struct is merged into `Config`.